framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -949,146 +951,175 @@ public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
+     * @param invokedMethodElement the ExecutableElement to retrieve the method preconditions from.
      */
-    protected void checkPreconditions(Tree tree,
-            Element invokedElement, boolean methodCall) {
-        checkPreconditions(tree, invokedElement, methodCall, null);
+    protected void checkMethodInvocationPreconditions(MethodInvocationTree methodInvocationTree,
+            ExecutableElement invokedMethodElement) {
+        if (invokedMethodElement == null) {
+            return;
+        }
+
+        checkPreconditions(methodInvocationTree, contractsUtils.getPreconditions(invokedMethodElement));
     }
 
     /**
-     * Checks all the preconditions of the method invocation or variable access {@code tree} with
-     * element {@code invokedElement}.
+     * Checks that all the given {@code preconditions} hold true immediately prior to
+     * the method invocation or variable access at {@code tree}.
+     *
+     * @param tree the Tree immediately prior to which the preconditions must hold true.
+     * @param preconditions the preconditions to be checked.
      */
-    protected void checkPreconditions(Tree tree,
-            Element invokedElement, boolean methodCall, Set<Pair<String, String>> additionalPreconditions) {
-        Set<Pair<String, String>> preconditions = invokedElement == null ?
-                new HashSet<Pair<String, String>>() :
-                contractsUtils.getPreconditions(invokedElement);
+    protected void checkPreconditions(Tree tree, Set<Pair<String, String>> preconditions) {
+        // This check is needed for the GUI effects and Units Checkers tests to pass.
+        // TODO: Remove this check and investigate the root cause.
+        if (preconditions.isEmpty()) {
+            return;
+        }
 
-        if (additionalPreconditions != null) {
-            preconditions.addAll(additionalPreconditions);
+        checkPreconditions(tree, atypeFactory.getNodeForTree(tree), preconditions);
+    }
+
+    /**
+     * Checks that all the given {@code preconditions} hold true immediately prior to
+     * the method invocation or variable access at {@code node}.  Errors are reported
+     * with respect to {@code tree}, which does not need to correspond to {@code node}.
+     *
+     * @param treeForErrorReporting the Tree used to report the error via checker.report.
+     * @param node the Node immediately prior to which the preconditions must hold true.
+     * @param preconditions the preconditions to be checked.
+     */
+    protected void checkPreconditions(Tree treeForErrorReporting,
+            Node node, Set<Pair<String, String>> preconditions) {
+        if (preconditions.isEmpty()) {
+            return;
         }
 
-        FlowExpressionContext flowExprContext = null;
+        FlowExpressionContext flowExprContext = getFlowExpressionContextFromNode(node);
+
+        if (flowExprContext == null) {
+            return;
```

Comment:
Isn't it an error if preconditions are present, but you can't determine the context? Aren't you possibly missing checks that should be performed?